src/main/java/org/mockito/internal/exceptions/Reporter.java

```diff
@@ -835,11 +835,45 @@ public static UnnecessaryStubbingException formatUnncessaryStubbingException(Cla
         for (Invocation u : unnecessaryStubbings) {
             stubbings.append("\n  ").append(count++).append(". ").append(u.getLocation());
         }
+        String heading = (testClass != null)?
+                "Unnecessary stubbings detected in test class: " + testClass.getSimpleName() :
+                "Unnecessary stubbings detected.";
+
         return new UnnecessaryStubbingException(join(
-                "Unnecessary stubbings detected in test class: " + testClass.getSimpleName(),
+                heading,
                 "Clean & maintainable test code requires zero unnecessary code.",
                 "Following stubbings are unnecessary (click to navigate to relevant line of code):" + stubbings,
                 "Please remove unnecessary stubbings or use 'silent' option. More info: javadoc for UnnecessaryStubbingException class."
         ));
     }
+
+    public static void unncessaryStubbingException(List<Invocation> unused) {
+        throw formatUnncessaryStubbingException(null, unused);
+    }
+
+    public static PotentialStubbingProblem potentialStubbingProblemByJUnitRule(Invocation actualInvocation, Collection<Invocation> matchingStubbings) {
+        StringBuilder stubbings = new StringBuilder();
+        int count = 1;
+        for (Invocation s : matchingStubbings) {
+            stubbings.append("  ").append(count++).append(". ").append(s);
+            stubbings.append("\n    ").append(s.getLocation()).append("\n");
+        }
+        stubbings.deleteCharAt(stubbings.length()-1); //remove trailing end of line
+
+        throw new PotentialStubbingProblem(join(
+                "Strict JUnit rule detected stubbing argument mismatch.",
+                "This invocation of '" + actualInvocation.getMethod().getName() + "' method:",
+                "  " + actualInvocation,
+                "  " + actualInvocation.getLocation(),
+                "Has following stubbing(s) with different arguments:",
+                stubbings,
+                "Typically, stubbing argument mismatch indicates user mistake when writing tests.",
+                "In order to streamline debugging tests Mockito fails early in this scenario.",
+                "However, there are legit scenarios when this exception generates false negative signal:",
+                "  - stubbing the same method multiple times using 'given' or 'when' syntax",
+                "    Please use willReturn/doReturn API for stubbing",
+                "  - stubbed method is intentionally invoked with different arguments by code under test",
+                "    Please use 'default' or 'silent' JUnit Rule.",
```

Comment:
e.g. with pseudo code

```
class TheTest {

MockitoRule.strictStubs() // current naming

  test1 {
    // works ok with strict stubs
  }
  test2 {
    // works ok with strict stubs
  }
  test3 {
    // incompatible with strict stubs 
  }
}
```

as test 3 is incompatible the solution is either to move it to another class, or to _silence_ the rule. In the later case, once silenced maintainance or evolution tasks will not benefit from the strict stubs, because the rule will be silent (or something else).

While the interesting part of quality is in maintaining the codebase. My idea would be to have the ability to deactivate by test, e.g.

```
  test3 {
    rule.deactivate()
    // incompatible with strict stubs 
  }
```